From 5bc5b10e001e55c3e3ff08ada9bc63ba167d4409 Mon Sep 17 00:00:00 2001
From: Ander Conselvan de Oliveira <ander.conselvan.de.oliveira@intel.com>
Date: Wed, 20 May 2015 09:03:27 +0300
Subject: [PATCH 0390/4706] drm/i915: Update comment in
 clear_intel_crtc_state()

commit 7546a38411fc3eb9e0ff1252718b0bf6579c87f6 upstream

Explain why a few fields of the new pipe_config have their values
preserved, while the others are zeroed.

Signed-off-by: Ander Conselvan de Oliveira <ander.conselvan.de.oliveira@intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
---
 drivers/gpu/drm/i915/intel_display.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 9d2d6fb..4a34709 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11485,7 +11485,11 @@ clear_intel_crtc_state(struct intel_crtc_state *crtc_state)
 	enum intel_dpll_id shared_dpll;
 	uint32_t ddi_pll_sel;
 
-	/* Clear only the intel specific part of the crtc state excluding scalers */
+	/* FIXME: before the switch to atomic started, a new pipe_config was
+	 * kzalloc'd. Code that depends on any field being zero should be
+	 * fixed, so that the crtc_state can be safely duplicated. For now,
+	 * only fields that are know to not cause problems are preserved. */
+
 	tmp_state = crtc_state->base;
 	scaler_state = crtc_state->scaler_state;
 	shared_dpll = crtc_state->shared_dpll;
-- 
1.7.5.4

